Packs/ApiModules/Scripts/TAXII2ApiModule/TAXII2ApiModule.py

```diff
@@ -1210,46 +1227,52 @@ def load_stix_objects_from_envelope(self, envelopes: types.GeneratorType, limit:
     def parse_generator_type_envelope(self, envelopes: types.GeneratorType, parse_objects_func, limit: int = -1):
         indicators = []
         relationships_lst = []
-        for envelope in envelopes:
-            stix_objects = envelope.get("objects")
-            if not stix_objects:
-                # no fetched objects
-                break
-
-            # we should build the id_to_object dict before iteration as some object reference each other
-            self.id_to_object.update(
-                {
-                    obj.get('id'): obj for obj in stix_objects
-                    if obj.get('type') not in ['extension-definition', 'relationship']
-                }
-            )
-
-            # now we have a list of objects, go over each obj, save id with obj, parse the obj
-            for obj in stix_objects:
-                obj_type = obj.get('type')
-
-                # we currently don't support extension object
-                if obj_type == 'extension-definition':
-                    continue
-                elif obj_type == 'relationship':
-                    relationships_lst.append(obj)
-                    continue
-
-                if not parse_objects_func.get(obj_type):
-                    demisto.debug(f'There is no parsing function for object type {obj_type}, '
-                                  f'available parsing functions are for types: {",".join(parse_objects_func.keys())}.')
-                    continue
-                if result := parse_objects_func[obj_type](obj):
-                    indicators.extend(result)
-                    self.update_last_modified_indicator_date(obj.get("modified"))
-
-                if reached_limit(limit, len(indicators)):
-                    return indicators, relationships_lst
+        try:
+            for envelope in envelopes:
+                stix_objects = envelope.get("objects")
+                if not stix_objects:
+                    # no fetched objects
+                    break
 
+                # we should build the id_to_object dict before iteration as some object reference each other
+                self.id_to_object.update(
+                    {
+                        obj.get('id'): obj for obj in stix_objects
+                        if obj.get('type') not in ['extension-definition', 'relationship']
+                    }
+                )
+                # now we have a list of objects, go over each obj, save id with obj, parse the obj
+                for obj in stix_objects:
+                    obj_type = obj.get('type')
+
+                    # we currently don't support extension object
+                    if obj_type == 'extension-definition':
+                        continue
```

Comment:
Can you add a demisto.debug() log when we are getting an 'extension-definition' object type?